resteasy-spring-feature-pack/common/src/main/resources/layers/standalone/resteasy-spring/layer-spec.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:2.0" name="resteasy-spring">
+    <props>
+        <prop name="org.wildfly.rule.xml-path" value="/WEB-INF/web.xml,/web-app/listener/listener-class,org.jboss.resteasy.plugins.server.servlet.ResteasyBootstrap"/>
+    </props>
+    <packages>
+        <package name="org.jboss.resteasy.resteasy-spring"/>
+        <package name="org.springframework.spring"/>
```

Comment:
I'm trying to think if this should be optional. Typically I think Spring is brought into the deployment, not provided as a module. In fact, I could be wrong, but I think there are occasionally issues with providing it as a module in a multi-tenant environment. I could be wrong though.

Reply:
I'm not sure my understanding is correct or not, but if I commented out this line and the tests still work. I guess maybe it's the `resteasy-spring` feature pack already bring in the `spring` module: https://github.com/resteasy/resteasy-spring/blob/main/resteasy-spring-feature-pack/common/src/main/resources/modules/system/layers/base/org/springframework/spring/main/module.xml @jfdenise wdyt?

@jamezp Currently it seems the spring framework can't be removed from the module, because the `resteasy-spring` module relies on it, and the project bring-in spring jars seems not work with the module (maybe I don't know how to do it correctly.).

Reply:
If being optional here is an option, that is the correct way to handle this IMO. The `org.springframework.spring` module should definitely not be required. I'd have to look at how this works though as I'm not a layers expert.

Reply:
@jamezp , we can make this package optional. But what I don't get is that the module org.springframework.spring is actually brought by this feature-pack. It is a strong dependency of org.jboss.resteasy.resteasy-spring-web. So even if made optional, we can't exclude it. It is strongly required it seems.

Reply:
Okay, if it's a strong dependency then we might need to re-think how we do this. It shouldn't hold this PR up, but IMO it should be fully optional.

Reply:
If we make the dependency optional in org.jboss.resteasy.resteasy-spring-web, we could then fully remove this package dependency from the layer. User would be in charge to provision org.springframework.spring by other means. How one would provision this module? What does it brings (I am completely newbee...)? Do we need another Galleon layer? This layer would have specific rules that would be bound to the usage of  org.springframework.spring.

Reply:
We introduced the `org.springframework.spring` module when we migrated to using Galleon for provisioning. The module is not required for applications to work if the user includes Spring in their deployment. However, if they want to not include Spring in their deployment, the module would be required.

Maybe a spring layer does make the most sense. 

Again, we could merge this PR and fix the layers in a different PR. Whatever works best IMO.

Reply:
> We introduced the `org.springframework.spring` module when we migrated to using Galleon for provisioning. The module is not required for applications to work if the user includes Spring in their deployment. However, if they want to not include Spring in their deployment, the module would be required.

I never knew this :)

I thought including Spring in the deployment project won't work, because the `resteasy-spring` module needs `spring` module to work. That's the way modules work I thought.

I'll verify this.

> 
> Maybe a spring layer does make the most sense.
> 
> Again, we could merge this PR and fix the layers in a different PR. Whatever works best IMO.

Okay I'll revert the change to @jfdenise 's PR's original state and merge the PR.

Reply:
That's the difference with the `deployment` and `inmodule` tests. The `deployment` tests are run without the module installed and Spring in the deployment. The `inmodule` tests are run without Spring in the deployment, but installed as a module. At least that is the intent.

Reply:
So I'm wrong and we are configuring the `org.springframework.spring` module in the test server. We shouldn't be doing that. We should also probably not be putting the module in the `modules/system/layers/base/org/springframework/spring/` directory and in `modules/org/springframework/spring/` instead.

That said, that is out of scope for this PR and something we should look at.

Reply:
Ah okay now I see the setup in the `inmodule` test:

```java
archive.addAsManifestResource(new StringAsset("Dependencies: org.springframework.spring meta-inf\n"), "MANIFEST.MF");
```

@jamezp Thanks for teaching!

Reply:
> So I'm wrong and we are configuring the `org.springframework.spring` module in the test server. We shouldn't be doing that. We should also probably not be putting the module in the `modules/system/layers/base/org/springframework/spring/` directory and in `modules/org/springframework/spring/` instead.
> 
> That said, that is out of scope for this PR and something we should look at.

@jamezp What's the difference between `modules/system/layers/base/org/springframework/spring/` and `modules/org/springframework/spring/`?

Reply:
@liweinan `system/layers/base` is supposed to be for WildFly defined modules. "Addons" or additional modules should go elsewhere. I've never been able to get it to work elsewhere though.